### ISSUE TITLE
Latest vkd3d-proton splits into d3d12.dll and d3d12core.dll yet prefix generation does not copy the file.

### ISSUE THREAD

**Issue by evenmynameistaken** [Frogging-Family/wine-tkg-git#978]
I'm not sure whose problem this is (upstream valve?) but recent vkd3d-proton master splits d3d12 into 2 dlls now and wine prefix generation and updating does not copy or update d3d12core.dll in the wine prefix.

I have been manually copying the appropriate ones into system32 and syswow64. I just did a new proton-tkg (valve bleeding edge) build a few minutes ago and see the problem remains (I checksummed the files to be sure)

I'm honestly not sure how Proton does this (from a list of files?), is it something I can fix for myself if it's something that's valve's responsibility (not proton-tkg)?

**Comment by evenmynameistaken** [comment]
> You could duplicate this code and replace d3d12 with d3d12core.

Thank you, that was the tidbit I needed. I didn't think to look at the proton wrapper itself :-)

**Comment by evenmynameistaken** [comment]
I got to thinking I should show what I did in case someone else needs a quick fix. I'm sure I'm not the only one clumsy with python (v.s. shell scripting I mean)
[proton-tkg-d3d12coredll.txt](https://github.com/Frogging-Family/wine-tkg-git/files/11114183/proton-tkg-d3d12coredll.txt)

(It won't let me attach as ".patch" even though listed lol)

**Comment by evenmynameistaken** [comment]
I'd say scrap that, it would fail in the event that a build is NOT going to use current vkd3d-proton master. I got looking at exactly what try_copy() does, it's defined in the proton script and will raise an exception if not "optional=True" so for now I added that to the lines. I don't know what else to do other than resorting to using wildcards/copying directory contents. But this should be fine if anyone uses it for a quick fix I'd think, while the change is still new.

[proton-tkg-d3d12coredll_optional.txt](https://github.com/Frogging-Family/wine-tkg-git/files/11115547/proton-tkg-d3d12coredll_optional.txt)

```diff -Naur wine-tkg-git_orig/proton-tkg/proton_template/conf/proton wine-tkg-git/proton-tkg/proton_template/conf/proton
--- wine-tkg-git_orig/proton-tkg/proton_template/conf/proton	2023-03-30 12:57:32.863367933 -0400
+++ wine-tkg-git/proton-tkg/proton_template/conf/proton	2023-03-30 15:46:28.975001324 -0400
@@ -1134,12 +1134,18 @@
                 if not os.path.exists(self.prefix_dir + "drive_c/windows/syswow64"):
                     try_copy(g_proton.lib_dir + "wine/vkd3d-proton/d3d12.dll",
                             self.prefix_dir + "drive_c/windows/system32/d3d12.dll")
+                    try_copy(g_proton.lib_dir + "wine/vkd3d-proton/d3d12core.dll",
+                            self.prefix_dir + "drive_c/windows/system32/d3d12core.dll", optional=True)
                     g_session.dlloverrides["d3d12"] = "n"
                 else:
                     try_copy(g_proton.lib64_dir + "wine/vkd3d-proton/d3d12.dll",
                             self.prefix_dir + "drive_c/windows/system32/d3d12.dll")
                     try_copy(g_proton.lib_dir + "wine/vkd3d-proton/d3d12.dll",
                             self.prefix_dir + "drive_c/windows/syswow64/d3d12.dll")
+                    try_copy(g_proton.lib64_dir + "wine/vkd3d-proton/d3d12core.dll",
+                            self.prefix_dir + "drive_c/windows/system32/d3d12core.dll", optional=True)
+                    try_copy(g_proton.lib_dir + "wine/vkd3d-proton/d3d12core.dll",
+                            self.prefix_dir + "drive_c/windows/syswow64/d3d12core.dll", optional=True)
                     g_session.dlloverrides["d3d12"] = "n"
 
             setup_game_dir_drive()
```

**Comment by Tk-Glitch** [comment]
Fixed with efeefff788e948b9a2c1d665c05e60025581df24

**Comment by Tk-Glitch** [comment]
We could set it to optional indeed for legacy compat. Using current being the default and it bringing quite a few enhancements should be preferred anyway.
